commitments/op_commitment.go

```diff
@@ -0,0 +1,90 @@
+package commitments
+
+import (
+	"fmt"
+	"log"
+)
+
+type OPCommitmentType byte
+
+const (
+	// Keccak256Byte represents a commitment using Keccak256 hashing.
+	Keccak256Byte OPCommitmentType = 0
+	// DAServiceByte represents a commitment using a DA service.
+	DAServiceByte OPCommitmentType = 1
+)
+
+type OPCommitment struct {
+	keccak256Commitment []byte
```

Comment:
What's the rationale for supporting this commitment type?

Reply:
We may have to support it in the future for chains migrating from AltLayer's keccak256-based EigenDA PoC.

Reply:
gotcha